### PR TITLE
B2CA-1684: Port Flex

### PR DESCRIPTION
**Pull request by cedelavergne-ledger**
Flex porting:

- Migrate to nbglv2
- Update version
- Add Flex config

# Checklist

- [X] App update process has been followed 
- [X] Target branch is `develop` 
- [X] Application version has been bumped

### CODE REVIEW

File: src/ui_nbgl.c [hunk 1]
```diff
@@ -20,107 +21,97 @@ static const char* const infoContents[] = {
     VERSION
 };
 
+static nbgl_layoutSwitch_t switches[3] = {0};
```
Comment (tdejoigny-ledger) [inline src/ui_nbgl.c:24]: use a #define for the switches number (will be reused after)

File: src/ui_nbgl.c [hunk 2]
```diff
@@ -20,107 +21,97 @@ static const char* const infoContents[] = {
...
+    switches[2] = (nbgl_layoutSwitch_t) {
+        .initState = (N_data.contract_data_type == ALLOW_CONTRACT_DATA) ? ON_STATE : OFF_STATE,
+        .text = "Allow contract data",
+        .subText = NULL,
+        .token = CONTRACT_DATA_TOKEN,
+        .tuneId = TUNE_TAP_CASUAL
+    };
+
+    content.type = SWITCHES_LIST;
+    content.content.switchesList.nbSwitches = 3;
```
Comment (tdejoigny-ledger) [inline src/ui_nbgl.c:85]: use a #define

**Review by tdejoigny-ledger — APPROVED** [review]
could we, at least, remove the deprecated `CX_APILEVEL `constant (and the linked code) ?

File: src/ui_nbgl.c (hunk 2 above)
Comment (cedelavergne-ledger) [reply src/ui_nbgl.c]: Done

File: src/ui_nbgl.c (hunk 1 above)
Comment (cedelavergne-ledger) [reply src/ui_nbgl.c]: Done

**Comment by cedelavergne-ledger** [comment]
Merging this 1st pass to allow the external dev team to fix remaining Tests issues (and deprecated API calls :wink: )

**Pull request merged**